oci/layer/unpack.go

```diff
@@ -120,19 +117,50 @@ func UnpackManifest(ctx context.Context, engine cas.Engine, bundle string, manif
 		return errors.Wrap(err, "bundle path empty")
 	}
 
+	defer func() {
+		if err != nil {
+			fsEval := fseval.DefaultFsEval
```

Comment:
AFAICS this logic is wrong (if you intended to not use `unpriv.RemoveAll` you could just do `os.RemoveAll`). You need to do something like in `UnpackRootfs`:

Reply:
You're right, I think I was misunderstanding why this particular is cleanup is needed.
Are you ok with a little duplication between the two function, or should we put that in a helper function `removeAll`?

Reply:
I'd just duplicate it, personally.